### PR TITLE
Disallow quick save/load during screen transition

### LINKED ISSUE #2191: Quickloading during a cutscene freezes the game
When pressing F9 during the "new game"-cutscene the game loads the savegame, but a frame of the cutscene still takes up the screen

### PR DESCRIPTION
**Pull request by Kasterisk**
**Brief Description of What This PR Does**

Added property in SaveHelper to allow and disallow quick save/load which is then used in TransitionManager to disallow performing those actions while transition is in progress.

**Related Issues**

Fixes #2191

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/saving/SaveHelper.cs [hunk 1]
```diff
@@ -41,6 +41,27 @@ public enum SaveOrder
         FileSystem,
     }
 
+    /// <summary>
+    ///   The error that is returned after performing quick load.
```
Comment (hhyyrylainen) [inline src/saving/SaveHelper.cs:45]:
```suggestion
    ///   The error that is returned after trying to perform a quick load.
```
?

File: src/saving/SaveHelper.cs [hunk 2]
```diff
@@ -41,6 +41,27 @@ public enum SaveOrder
...
+    /// </summary>
+    public enum QuickLoadError
+    {
+        /// <summary>
+        ///   No error, quick load is successful (or error is not handled).
+        /// </summary>
+        None,
+
+        /// <summary>
+        ///   The loaded version do not match the current game version.
```
Comment (hhyyrylainen) [inline src/saving/SaveHelper.cs:55]:
```suggestion
        ///   The loaded version does not match the current game version.
```

File: src/saving/SaveHelper.cs [hunk 3]
```diff
@@ -41,6 +41,27 @@ public enum SaveOrder
...
+        /// </summary>
+        VersionMismatch,
+
+        /// <summary>
+        ///   Quick load is made not allowed to be performed.
```
Comment (hhyyrylainen) [inline src/saving/SaveHelper.cs:60]:
```suggestion
        ///   Quick load is currently prevented by the current state of the game.
```
I think this is slightly better wording, but not perfect either

File: src/saving/SaveHelper.cs [hunk 4]
```diff
@@ -52,6 +73,8 @@ public enum SaveOrder
     /// <returns>True if the last save is still recent, false if otherwise.</returns>
     public static bool SavedRecently => lastSave != null ? DateTime.Now - lastSave < Constants.RecentSaveTime : false;
 
+    public static bool AllowQuickSavingAndLoading { get; set; } = true;
```
Comment (hhyyrylainen) [inline src/saving/SaveHelper.cs:76]: Please also document here what this property does.

File: src/saving/SaveHelper.cs [hunk 5]
```diff
@@ -81,6 +104,9 @@ public static void Save(string name, MicrobeEditor state)
     /// <param name="state">Data to include in save</param>
     public static void QuickSave(MicrobeStage state)
     {
+        if (!AllowQuickSavingAndLoading)
```
Comment (hhyyrylainen) [inline src/saving/SaveHelper.cs:107]: Maybe InternalSaveHelper should check this? As we'll end up with 16 quick save variants (at least, one per stage and editor, and with much more editors later in the game, we might end up with a ton of these checks being duplicated).

Comment (Kasterisk) [reply src/saving/SaveHelper.cs:107]: Yeah that's preferrable, I actually didn't saw that there's a SaveType parameter that you can use to distinguish saves, that was why I done it like this.

**Review by RacerBG — APPROVED** [review]
Nothing happens when I press F5 or F9 during the cutscene, I believe this is the expected behaviour of the game.

File: src/saving/SaveHelper.cs [hunk 6]
```diff
@@ -398,6 +427,9 @@ public static void ClearLastSaveTime()
     private static void InternalSaveHelper(SaveInformation.SaveType type, MainGameState gameState,
         Action<Save> copyInfoToSave, Func<Node> stateRoot, string? saveName = null)
     {
+        if (type == SaveInformation.SaveType.QuickSave && !AllowQuickSavingAndLoading)
+            return;
```
Comment (hhyyrylainen) [inline src/saving/SaveHelper.cs:431]: One more tiny thing, could you add a print here warning the user why they couldn't save? That way if ever AllowQuickSavingAndLoading variable has a bug and isn't reset, we can see from the game logs why someone complains they can't save.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/saving/SaveHelper.cs (hunk 6 above)
Comment (hhyyrylainen) [reply src/saving/SaveHelper.cs:431]: To clarify it should be something like `GD.Print("Can't save due to quick save being currently suppressed");`

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**